lib/runtime-c-api/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+    //    let params= &[Value::I32(3), Value::I32(4)];
+
+    let func_name_c = unsafe { CStr::from_ptr(name) };
+    let func_name_r = func_name_c.to_str().unwrap();
+    let instance = unsafe { Box::from_raw(instance as *mut Instance) };
+
+    let results: &mut [wasmer_value_t] = slice::from_raw_parts_mut(results, results_len as usize);
+    let result = instance.call(func_name_r, &params[..]);
+    Box::into_raw(instance);
+    match result {
+        Ok(results_vec) => {
+            println!("Call Res: {:?}", results_vec);
+            if results_vec.len() > 0 {
+                let ret = match results_vec[0] {
+                    Value::I32(x) => wasmer_value_t {
+                        tag: wasmer_value_tag::WASM_I32,
+                        value: wasmer_value { I32: x },
+                    },
+                    Value::I64(x) => wasmer_value_t {
+                        tag: wasmer_value_tag::WASM_I64,
+                        value: wasmer_value { I64: x },
+                    },
+                    Value::F32(x) => wasmer_value_t {
+                        tag: wasmer_value_tag::WASM_F32,
+                        value: wasmer_value { F32: x },
+                    },
+                    Value::F64(x) => wasmer_value_t {
+                        tag: wasmer_value_tag::WASM_F64,
+                        value: wasmer_value { F64: x },
+                    },
+                };
+                results[0] = ret;
+            }
+            wasmer_call_result_t::WASMER_CALL_OK
+        }
+        Err(err) => {
+            println!("Call Err: {:?}", err);
+            wasmer_call_result_t::WASMER_CALL_ERROR
+        }
+    }
+}
+
+#[allow(clippy::cast_ptr_alignment)]
+#[no_mangle]
+pub unsafe extern "C" fn wasmer_imports_set_import_func(
+    import_object: *mut wasmer_import_object_t,
+    namespace: *const c_char,
+    name: *const c_char,
+    func: extern "C" fn(data: *mut c_void),
+    params: *const wasmer_value_tag,
+    params_len: c_int,
+    returns: *const wasmer_value_tag,
+    returns_len: c_int,
+) {
+    let mut import_object = unsafe { Box::from_raw(import_object as *mut ImportObject) };
+    let namespace_c = unsafe { CStr::from_ptr(namespace) };
+    let namespace_r = namespace_c.to_str().unwrap();
+    let name_c = unsafe { CStr::from_ptr(name) };
+    let name_r = name_c.to_str().unwrap();
+
+    let params: &[wasmer_value_tag] = slice::from_raw_parts(params, params_len as usize);
+    let params: Vec<Type> = params.iter().cloned().map(|x| x.into()).collect();
+    let returns: &[wasmer_value_tag] = slice::from_raw_parts(returns, returns_len as usize);
+    let returns: Vec<Type> = returns.iter().cloned().map(|x| x.into()).collect();
+
+    let export = Export::Function {
+        func: unsafe { FuncPointer::new(func as _) },
+        ctx: Context::Internal,
+        signature: Arc::new(FuncSig::new(params, returns)),
+    };
+
+    // TODO handle existing namespace
```

Comment:
Yeah, unfortunately, the way namespaces currently work doesn't allow this. We could either change the implementation, or the user could make a `namespace_object`, fill that with imports, and then import that into the `import_object`?

Reply:
I think I'll update this to add a `namespace_object` to be inline with the current rust API.

Reply:
I've updated instantiate to take an array of imports so this issue does not happen since I don't need to mutate these.